src/test/java/bio/overture/ego/controller/PolicyControllerTest.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2019. The Ontario Institute for Cancer Research. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package bio.overture.ego.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import bio.overture.ego.AuthorizationServiceMain;
+import bio.overture.ego.model.entity.Policy;
+import bio.overture.ego.service.PolicyService;
+import bio.overture.ego.utils.EntityGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@Slf4j
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = AuthorizationServiceMain.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class PolicyControllerTest {
+
+  /** Constants */
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  /** State */
+  @LocalServerPort private int port;
+
+  private TestRestTemplate restTemplate = new TestRestTemplate();
+  private HttpHeaders headers = new HttpHeaders();
+  private static boolean hasRunEntitySetup = false;
+
+  /** Dependencies */
+  @Autowired private EntityGenerator entityGenerator;
+
+  @Autowired PolicyService policyService;
+
+  @Before
+  public void setup() {
+    // Initial setup of entities (run once
+    if (!hasRunEntitySetup) {
+      entityGenerator.setupTestUsers();
+      entityGenerator.setupTestGroups();
+      entityGenerator.setupTestPolicies();
+      hasRunEntitySetup = true;
+    }
+
+    headers.add("Authorization", "Bearer TestToken");
+    headers.setContentType(MediaType.APPLICATION_JSON);
+  }
+
+  @Test
+  @SneakyThrows
+  public void addPolicy() {
+    val policy = Policy.builder().name("AddPolicy").build();
```

Comment:
Since you are creating a new policy, the 
`PolicyRequest` DTO should be used and not the Policy entity. So replace with
`PolicyRequest.builder().name("AddPolicy").build()`

although its a transparent change, it makes it clear to future developers what the required DTO is